### PR TITLE
feat: verify if passed txid is mined for mint! and burn!

### PR DESCRIPTION
**Pull request by MarvinJanssen**
Addresses #79, but the exact purpose of the requirement remains unclear. Since only the contract owner can `mint!` and `burn!`, and these functions will not be part of the API, why is the check truly needed here?

Think this is enough @friedger @FriendsFerdinand? I'll add tests if you agree with the approach.

### CODE REVIEW

**Comment by friedger** [comment]
Looks good, this approach. The verification is needed so that we know that sbtc is minted on the stacks chain using the same bitcoin fork as the bitcoin node.

File: romeo/asset-contract/contracts/asset.clar [hunk 1]
```diff
@@ -5,16 +5,16 @@
 
 ;; traits
 ;;
-(impl-trait 'ST1NXBK3K5YYMD6FD41MVNP3JS1GABZ8TRVX023PT.sip-010-trait-ft-standard.sip-010-trait)
+(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)
```
Comment (jo-tm) [inline romeo/asset-contract/contracts/asset.clar:8]: Do these traits are only in Mainnet or can be deployed in Devnet with the same principal address?

**Review by jo-tm — APPROVED** [review]
approved with a minor question on hardoced principals...

File: romeo/asset-contract/contracts/asset.clar (hunk 1 above)
Comment (friedger) [reply romeo/asset-contract/contracts/asset.clar:8]: They can be removed alltogether

**Comment by friedger** [comment]
@MarvinJanssen Can you use branch 79-proof as your base? It contains the changes to call your function.

**Review by FriendsFerdinand — APPROVED** [review]

**Review by friedger — APPROVED** [review]
LGTM

Tests still open.

File: romeo/asset-contract/contracts/asset.clar [hunk 2]
```diff
@@ -90,5 +118,9 @@
 ;; private functions
 ;;
 (define-private (is-contract-owner)
-    (is-eq (var-get contract-owner) tx-sender)
+    (ok (asserts! (is-eq (var-get contract-owner) contract-caller) err-invalid-caller))
```
Comment (friedger) [inline romeo/asset-contract/contracts/asset.clar:121]: Is this more efficient?

For a return ok, I would expect the name of the function something like `check-is-contract-owner`.

Comment (MarvinJanssen) [reply romeo/asset-contract/contracts/asset.clar:121]: Not sure if it is more efficient but it definitely makes using the function easier to use (with a `try!`)—and there is no chance you return the wrong error value somewhere. It is a standard pattern I follow.

**Comment by MarvinJanssen** [comment]
I'll do tests in the next PR.

**Comment by MarvinJanssen** [comment]
@friedger I didn't notice you changed the base so now I merged it into `79-proof` instead of `master`. I'm not sure how to undo it on Github.

**Pull request merged**